stan/math/prim/fun/select.hpp

```diff
@@ -0,0 +1,176 @@
+#ifndef STAN_MATH_PRIM_FUN_SELECT_HPP
+#define STAN_MATH_PRIM_FUN_SELECT_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/err.hpp>
+#include <stan/math/prim/functor/apply_scalar_binary.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return the second argument if the first argument is true
+ * and otherwise return the third argument.
+ *
+ * <code>select(c, y1, y0) = c ? y1 : y0</code>.
+ *
+ * @tparam T_true type of the true argument
+ * @tparam T_false type of the false argument
+ * @param c Boolean condition value.
+ * @param y_true Value to return if condition is true.
+ * @param y_false Value to return if condition is false.
+ */
+template <typename T_true, typename T_false,
+          typename ReturnT = return_type_t<T_true, T_false>,
+          require_all_stan_scalar_t<T_true, T_false>* = nullptr>
+inline ReturnT select(const bool c, const T_true y_true,
+                      const T_false y_false) {
+  return c ? ReturnT(y_true) : ReturnT(y_false);
+}
```

Comment:
`ReturnT` not doc'd here

Reply:
`ReturnT` is a derived type (the result of `return_type_t<...>`), and we haven't usually doc'd them AFAIK